iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -251,6 +254,21 @@ public void commitAlterTable(org.apache.hadoop.hive.metastore.api.Table hmsTable
       }
       HiveTableUtil.importFiles(preAlterTableProperties.tableLocation, preAlterTableProperties.format,
           partitionSpecProxy, preAlterTableProperties.partitionKeys, catalogProperties, conf);
+    } else {
+      Map<String, String> contextProperties = context.getProperties();
+      if (contextProperties.containsKey(ALTER_TABLE_OPERATION_TYPE) &&
+          allowedAlterTypes.contains(contextProperties.get(ALTER_TABLE_OPERATION_TYPE))) {
+        Map<String, String> hmsTableParameters = hmsTable.getParameters();
+        Splitter splitter = Splitter.on(PROPERTIES_SEPARATOR);
+        UpdateProperties icebergUpdateProperties = icebergTable.updateProperties();
+        if (contextProperties.containsKey(SET_PROPERTIES)) {
+          splitter.splitToList(contextProperties.get(SET_PROPERTIES))
+              .forEach(k -> icebergUpdateProperties.set(k, hmsTableParameters.get(k)));
+        } else {
```

Comment:
maybe make this `else if (contextProperties.containsKey(UNSET_PROPERTIES))`? We might later call this method from other alter table ops (e.g. schema change) and we wouldn't want to enter this block then